docs/auto-instrumentation-install.md

```diff
@@ -0,0 +1,282 @@
+
+# If cert-manager is already deployed.
+helm install splunk-otel-collector -f ./my_values.yaml --set operator.enabled=true -n monitoring helm-charts/splunk-otel-collector
+```
+
+### 2. Deploy the opentelemetry.io/v1alpha1 Instrumentation
+
+This Instrumentation object is a spec to configure what instrumentation libraries to use for instrumentation. An
+Instrumentation object must be available to the target pod for auto-instrumentation to function. Here is an example
+of a Instrumentation yaml file and how to install it.
+
+<details open>
+<summary>Expand for sample splunk-instrumentation.yaml</summary>
+
+```yaml
+# splunk-instrumentation.yaml
+apiVersion: opentelemetry.io/v1alpha1
+kind: Instrumentation
+metadata:
+  name: splunk-instrumentation
+spec:
+  exporter:
+    endpoint: http://$(SPLUNK_OTEL_AGENT):4317
+  propagators:
+    - tracecontext
+    - baggage
+    - b3
+  java:
+    env:
+      - name: SPLUNK_OTEL_AGENT
+        valueFrom:
+          fieldRef:
+            apiVersion: v1
+            fieldPath: status.hostIP
+      - name: OTEL_EXPORTER_OTLP_ENDPOINT
+        value: https://$(SPLUNK_OTEL_AGENT):4318
+      - name: OTEL_TRACES_EXPORTER
+        value: otlp
+  dotnet:
+    image: your-customized-auto-instrumentation-image:dotnet
+    env:
+      - name: OTEL_EXPORTER_OTLP_ENDPOINT
+        value: http://{TARGET}:4317
+  # nodjs:
+  # python:
+```
+
+</details>
+
+```
+# Install
+kubectl apply -f splunk-instrumentation.yaml
+# Check the current deployed values
+kubectl get otelint -o yaml
+```
+
+### 3. Verify all the OpenTelemetry resources (collector, operator, webhook, instrumentation) are deployed successfully
+
+<details open>
+<summary>Expand for sample output to verify against</summary>
+
+```
+kubectl  get pods -n monitoring
+# NAME                                                          READY
+# NAMESPACE     NAME                                                            READY   STATUS
+# monitoring    splunk-otel-collector-agent-lfthw                               2/2     Running
+# monitoring    splunk-otel-collector-cert-manager-6b9fb8b95f-2lmv4             1/1     Running
+# monitoring    splunk-otel-collector-cert-manager-cainjector-6d65b6d4c-khcrc   1/1     Running
+# monitoring    splunk-otel-collector-cert-manager-webhook-87b7ffffc-xp4sr      1/1     Running
+# monitoring    splunk-otel-collector-k8s-cluster-receiver-856f5fbcf9-pqkwg     1/1     Running
+# monitoring    splunk-otel-collector-opentelemetry-operator-56c4ddb4db-zcjgh   2/2     Running
+
+kubectl get mutatingwebhookconfiguration.admissionregistration.k8s.io -n monitoring
+# NAME                                      WEBHOOKS   AGE
+# splunk-otel-collector-cert-manager-webhook              1          14m
+# splunk-otel-collector-opentelemetry-operator-mutation   3          14m
+
+kubectl get otelinst -n spring-petclinic
+# NAME                          AGE   ENDPOINT
+# splunk-instrumentation        3m   http://$(SPLUNK_OTEL_AGENT):4317
```

Comment:
This seems unexpected

Reply:
- Instrumentation is a namespaced object we have to store some where. Ideally we would store one Instrumentation instance in the namespace the collector is installed in. Then auto-instrumentation happening in other namespaces should reference our one default Instrumentation instance .
- The `instrumentation.opentelemetry.io/inject-{library}: "namespace-name/instrumentation-name"` annotation  is supposed to allow us to use an Instrumentation instances from different namespaces. Last week I was digging into this and I did find a bug that relates to this. The cross namespace annotation is only respected in a subset of cases it should be respected. Until this bug is addressed, the work around is we will have to create an Instrumentation object in every namespace that will be using auto-instrumentation.

Reply:
Should this be `spring-petclinic` namespace? This instruction doesn't mention spring-petclinic prior to this

Reply:
Yeah it should be spring-petclinic for now. We want the collector and operator to reside in a namespace our target applications are not located in. I added more notes to the install instructions. 

Hopefully in the future we should be able to support better cross namespace access of Instrumentation objects.

Reply:
Not sure I understand why this namespace is used here. This document instructs users to install collector in `monitoring` and instrumentation in the default namespace (should they be the same?), but no mentions of `spring-petclinic` up to this point

Reply:
Updated docs

Reply:
How did you update the docs?

We still ask user to deploy instrumentation in the [default namespace](https://github.com/signalfx/splunk-otel-collector-chart/pull/701/files/5dddb1579d4a929077a9ed241228cf9373071e42#diff-831285ac5240fe6b18737811f1bb2387e71261c895f92c090a8878e10d7e22d7R90):

```
kubectl apply -f splunk-instrumentation.yaml
```

but here look for it in `spring-petclinic`

```
kubectl get otelinst -n spring-petclinic
```

Reply:
I did miss adding the "-n spring-petclinic" to the apply command. Updated.

Reply:
Still unclear why do we install it in `spring-petclinic` while the collector is installed in `monitoring`?

Reply:
- I want this example to demonstrate how to auto-instrument applications that do not reside within the same namespace as the operator.
- There is a bug or undocumented feature that prevents application pods from being auto-instrumented using an Instrumentation object that does not reside within the same namespace. I've been digging into.
  - Observed Behaviors
    - The application and operator pods exist in namespace X as well as the instrumentation object exist in namespace Y -> Auto-instrumentation succeeds
    - The application pods and instrumentation object exist in namespace X as well as the operator pods exist in namespace Y -> Auto-instrumentation succeeds
    - The application pods exist in namespace X as well as the operator pods and instrumentation object exist in namespace Y -> Auto-instrumentation fails (application cannot find Instrumentation)

Reply:
Ok, thanks for clarifying